client/src/__tests__/unit-tests/bitbake-driver.test.ts

```diff
@@ -0,0 +1,60 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2023 Savoir-faire Linux. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as fs from 'fs'
+import * as vscode from 'vscode'
+import { BitbakeDriver } from '../../driver/BitbakeDriver'
+
+jest.mock('vscode')
+
+const mockBitbakeConfiguration = (values: Record<string, string>): void => {
+  vscode.workspace.getConfiguration = jest.fn().mockImplementation(() => ({
+    get: (key?: string): string | undefined => {
```

Comment:
Mocks are exempt from the Typescript typing checks because they are pure javascript chaos at runtime so signature is not an issue unless the actual arguments do not have the expected prototypes. Hence I made key mandatory because the test always uses it anyway.

Reply:
ye right, just the expressions in `bitbakesettings` are invoking the one with the second parameter. Though I don't really see any big issues in this case, generally it is not wrong to have a mock that is as close to the real one as possible. Anyways, I won't block the PR on this.